homeassistant/components/tuya/vacuum.py

```diff
@@ -184,3 +186,9 @@ def locate(self, **kwargs: Any) -> None:
     def set_fan_speed(self, fan_speed: str, **kwargs: Any) -> None:
         """Set fan speed."""
         self._send_command([{"code": DPCode.SUCTION, "value": fan_speed}])
+
+    def send_command(
+        self, command: list[dict[str, Any]], params: str = None, **kwargs: Any
```

Comment:
I don't think `params` is a string?
The `command` isn't a list or dictionary either?

Where did you base this on? Maybe I am misunderstanding it?

Reply:
The `command` was based on [Tuya's documentation](https://developer.tuya.com/en/docs/cloud/e2512fb901?id=Kag2yag3tiqn5#title-3-Request%20example) and on the Explorer API, where a command with multiple codes can be sent to the same device.


````
{
  "commands": [
    {
      "code": "volume_set",
      "value": 100
    },
    {
      "code": "seek",
      "value": true
    }
  ]
}
````

In the home assistant, the service would look like this:
````
service: vacuum.send_command
data:
  command: '[{"code": "volume_set","value": 100},{"code": "seek","value": true}]'
target:
  device_id: 1694986f989da663be363592a70642d9
````

The `params` has no function for the send_command service to vacuum Tuya.

Reply:
It doesn't matter is parameters is used or not, it still needs to be typed correctly.

For command, it doesn't matter what the Tuya docs say, this is a home assistant service, and it home assistant will call this method. It must be typed towards internal use, not external Tuya use.

Looking at your service example, y9ir are not putting in a dictionary, yet, your have typed it as such.